### PR TITLE
refactor: stop encoding plantree metadata into rendered strings

### LINKED ISSUE #15: plantree: stop encoding row metadata into rendered tree strings
## Problem
The current `plantree.ProcessPlan` / `buildTree` pipeline passes structured node metadata through `treeprint` by serializing it into the rendered string, then parsing it back out later.

Relevant code:
- `plantree/plantree.go:132-195`
- `plantree/plantree.go:205-226`

Today this works by embedding:
- JSON-encoded node text
- JSON-encoded child-link metadata
- `\t` separators
- a `\000` terminator
- extra prefixed newlines to compensate for wrapped node text

## Why this matters
This creates a tight coupling between:
- tree formatting
- wrapping behavior
- row extraction
- metadata transport

That makes the rendering pipeline brittle. Small changes to formatting or escaping rules can accidentally break row reconstruction with errors like `unexpected format, tree line = ...`.

It also makes future work on `rendertree` harder than necessary because metadata and presentation are not separated.

## Expected behavior
`plantree` should keep structured row metadata separate from the rendered tree string.

Possible directions:
1. Build a structured intermediate tree and only stringify at the final rendering step.
2. Store metadata in side tables keyed by node ID instead of encoding it into `treeprint` values.
3. Split the current `ProcessPlan` into a traversal phase and a formatting phase.

## Suggested direction
I would treat this as a refactor issue rather than a behavior change. The current output can stay the same, but the internal handoff should stop depending on JSON + tab + NUL encoding inside display strings.

### PR DESCRIPTION
**Pull request by apstndb**
## Summary
- replace the JSON + tab + NUL metadata handoff in `plantree.ProcessPlan` with a structured intermediate tree
- keep `RowWithPredicates`, `TreePart`, `NodeText`, and `WithTreeprintOptions` behavior compatible for downstream callers such as `spanner-mycli`
- add direct `plantree` regression tests for current formatting, wrap handling, and custom treeprint edge configuration

Closes #15.

## Verification
- `go test ./...`
- `golangci-lint run --timeout=5m`
- `go test ./internal/mycli -run 'TestRenderTreeUsingTestdataPlans|TestRenderTreeWithStats'` in `spanner-mycli` via a temporary `go.work` pointing at this branch

### CODE REVIEW

File: plantree/plantree.go [hunk 1]
```diff
@@ -123,88 +139,50 @@ func ProcessPlan(qp *spannerplan.QueryPlan, opts ...Option) (rows []RowWithPredi
...
 	}
 
-	tree := treeprint.New()
-
-	if err := buildTree(qp, tree, nil, 0, &o); err != nil {
-		return nil, fmt.Errorf("failed on buildTree, err: %w", err)
+	root, err := buildRenderedTree(qp, nil, 0, &o)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build rendered tree: %w", err)
 	}
```
[the remark this answers is not in the document]
Comment (apstndb) [reply plantree/plantree.go:145]: Addressed in 0fccb22. ProcessPlan now returns an empty result when buildRenderedTree yields no visible root, and I added TestProcessPlan_InvisibleRootReturnsEmpty to lock that behavior in.
